lib/json/ext.rb

```diff
@@ -2,24 +2,26 @@
 
 require 'json/common'
 
-module JSON
-  # This module holds all the modules/classes that implement JSON's
-  # functionality as C extensions.
-  module Ext
-    if RUBY_ENGINE == 'truffleruby'
-      require 'json/ext/parser'
-      require 'json/pure'
-      $DEBUG and warn "Using Ext extension for JSON parser and Pure library for JSON generator."
-      JSON.parser = Parser
-      JSON.generator = JSON::Pure::Generator
-    else
-      require 'json/ext/parser'
-      require 'json/ext/generator'
-      $DEBUG and warn "Using Ext extension for JSON."
-      JSON.parser = Parser
-      JSON.generator = Generator
-    end
+unless defined?(::JSON::JSON_LOADED)
+  # Prevent json/pure from being loaded at the same time
+  $LOADED_FEATURES << File.expand_path("../pure.rb", __FILE__)
```

Comment:
I tired to add some stub `json/ext/generator.rb` etc, but it was making loading the test suite very contrived.

In the end I think marking the alternative implementation as loaded is OK?

Reply:
Given that file doesn't exist (because of the changes in json.gemspec), does it have any effect?
It feels brittle.
Is it necessary? I would think the `JSON::JSON_LOADED` check is enough.

Reply:
It's in case the version of the "other gem" is much older.